Framework/DataObjects/src/Peak.cpp

```diff
@@ -177,6 +178,14 @@ Peak::Peak(const Geometry::IPeak &ipeak)
   }
 }
 
+//----------------------------------------------------------------------------------------------
+Peak::Peak(const Mantid::DataObjects::LeanElasticPeak &lpeak,
+           const Geometry::Instrument_const_sptr &inst,
+           boost::optional<double> detectorDistance) {
```

Comment:
Can you add in `: BasePeak(lpeak)` in order to copy over all the member variables in the `BasePeak`

Reply:
Thanks for pointing it out.  The constructor has been updated accordingly.